// emailjs-com/cjs/methods/send/send.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.send = void 0;

var store_1 = require("../../store/store");

var validateParams_1 = require("../../utils/validateParams");

var sendPost_1 = require("../../api/sendPost");
/**
 * Send a template to the specific EmailJS service
 * @param {string} serviceID - the EmailJS service ID
 * @param {string} templateID - the EmailJS template ID
 * @param {object} templatePrams - the template params, what will be set to the EmailJS template
 * @param {string} userID - the EmailJS user ID
 * @returns {Promise<EmailJSResponseStatus>}
 */


var send = function send(serviceID, templateID, templatePrams, userID) {
  var uID = userID || store_1.store._userID;
  validateParams_1.validateParams(uID, serviceID, templateID);
  var params = {
    lib_version: '3.2.0',
    user_id: uID,
    service_id: serviceID,
    template_id: templateID,
    template_params: templatePrams
  };
  return sendPost_1.sendPost('/api/v1.0/email/send', JSON.stringify(params), {
    'Content-type': 'application/json'
  });
};

exports.send = send;
